Looks like the build error is caused by something else:

    FAILED: zephyr/zephyr.ri /__w/zephyr/zephyr/twister-out/intel_adsp_cavs25/tests/boards/intel_adsp/ssp/boards.intel_adsp.ssp/zephyr/zephyr.ri 
    cd /__w/zephyr/zephyr/twister-out/intel_adsp_cavs25/tests/boards/intel_adsp/ssp/boards.intel_adsp.ssp/zephyr/soc/xtensa/intel_adsp/common && west sign --if-tool-available --tool rimage --build-dir /__w/zephyr/zephyr/twister-out/intel_adsp_cavs25/tests/boards/intel_adsp/ssp/boards.intel_adsp.ssp
    usage: west [-h] [-z ZEPHYR_BASE] [-v] [-V] <command> ...
    west: error: argument <command>: invalid choice: 'sign' (choose from 'init', 'update', 'list', 'manifest', 'diff', 'status', 'forall', 'help', 'config', 'topdir', 'selfupdate')
    ninja: build stopped: subcommand failed.

--- a/drivers/adc/adc_ads114s0x.c
+++ b/drivers/adc/adc_ads114s0x.c
@@ -9,6 +9,7 @@
 #include <zephyr/drivers/adc/ads114s0x.h>
 #include <zephyr/drivers/spi.h>
 #include <zephyr/drivers/gpio.h>
+#include <zephyr/dt-bindings/adc/ads114s0x_adc.h>
 #include <zephyr/logging/log.h>
 #include <zephyr/kernel.h>
 #include <zephyr/sys/__assert.h>
@@ -596,6 +597,8 @@ static int ads114s0x_channel_setup(const struct device *dev,
 	int result;
 	enum ads114s0x_register register_addresses[6];
 	uint8_t values[ARRAY_SIZE(register_addresses)];
+	uint16_t acquisition_time_value = ADC_ACQ_TIME_VALUE(channel_cfg->acquisition_time);
+	uint16_t acquisition_time_unit = ADC_ACQ_TIME_UNIT(channel_cfg->acquisition_time);
 
 	ADS114S0X_REGISTER_INPMUX_SET_DEFAULTS(gain);
 	ADS114S0X_REGISTER_REF_SET_DEFAULTS(reference_control);
@@ -607,6 +610,23 @@ static int ads114s0x_channel_setup(const struct device *dev,
 	if (channel_cfg->channel_id != 0) {
 		LOG_ERR("only one channel is supported");
 		return -EINVAL;
+	}
+
+	/* The ADS114 uses samples per seconds units with the lowest being 2.5SPS
+	 * and with acquisition_time only having 14b for time, this will not fit
+	 * within here for microsecond units. Use Tick units and allow the user to
+	 * specify the ODR directly.
+	 */
+	if (channel_cfg->acquisition_time != ADC_ACQ_TIME_DEFAULT &&
+	    acquisition_time_unit != ADC_ACQ_TIME_TICKS) {
+		LOG_ERR("invalid acquisition time %i", channel_cfg->acquisition_time);
+		return -EINVAL;
+	}
+
+	if (channel_cfg->acquisition_time == ADC_ACQ_TIME_DEFAULT) {
+		ADS114S0X_REGISTER_DATARATE_DR_SET(data_rate, ADS114S0X_CONFIG_DR_20);
+	} else {
+		ADS114S0X_REGISTER_DATARATE_DR_SET(data_rate, acquisition_time_value);
 	}
 
 	switch (channel_cfg->reference) {

--- a/include/zephyr/dt-bindings/adc/ads114s0x_adc.h
+++ b/include/zephyr/dt-bindings/adc/ads114s0x_adc.h
@@ -1,0 +1,29 @@
+/*
+ * Copyright (c) 2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_ADC_ADS114S0X_ADC_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_ADC_ADS114S0X_ADC_H_
+
+/*
+ * These are the available data rates described as samples per second. They
+ * can be used with the time unit ticks for the acquisition time.
+ */
+#define ADS114S0X_CONFIG_DR_2_5		0
+#define ADS114S0X_CONFIG_DR_5		1
+#define ADS114S0X_CONFIG_DR_10		2
+#define ADS114S0X_CONFIG_DR_16_6	3
+#define ADS114S0X_CONFIG_DR_20		4
+#define ADS114S0X_CONFIG_DR_50		5
+#define ADS114S0X_CONFIG_DR_60		6
+#define ADS114S0X_CONFIG_DR_100		7
+#define ADS114S0X_CONFIG_DR_200		8
+#define ADS114S0X_CONFIG_DR_400		9
+#define ADS114S0X_CONFIG_DR_800		10
+#define ADS114S0X_CONFIG_DR_1000	11
+#define ADS114S0X_CONFIG_DR_2000	12
+#define ADS114S0X_CONFIG_DR_4000	13
+
+#endif /* ZEPHYR_INCLUDE_DT_BINDINGS_ADC_ADS114S0X_ADC_H_ */

